gadgets/src/is_zero.rs

```diff
@@ -18,19 +26,41 @@ pub(crate) trait IsZeroInstruction<F: FieldExt> {
     ) -> Result<(), Error>;
 }
 
+/// Config struct representing the required fields for an `IsZero` config to
+/// exist.
 #[derive(Clone, Debug)]
-pub(crate) struct IsZeroConfig<F> {
+pub struct IsZeroConfig<F> {
+    /// Modular inverse of the value.
     pub value_inv: Column<Advice>,
     /// This can be used directly for custom gate at the offset if `is_zero` is
     /// called, it will be 1 if `value` is zero, and 0 otherwise.
     pub is_zero_expression: Expression<F>,
 }
 
-pub(crate) struct IsZeroChip<F> {
+/// Wrapper arround [`IsZeroConfig`] for which [`Chip`] is implemented.
+pub struct IsZeroChip<F> {
     config: IsZeroConfig<F>,
 }
 
 impl<F: FieldExt> IsZeroChip<F> {
+    #[rustfmt::skip]
+    /// Sets up the configuration of the chip by creating the required columns
+    /// and defining the constraints that take part when using `is_zero` gate.
+    ///
+    /// Truth table of iz_zero gate:
+    /// +----+-------+-----------+-----------------------+---------------------------------+-------------------------------------+
+    /// | ok | value | value_inv | 1 - value ⋅ value_inv | value ⋅ (1 - value ⋅
+    /// value_inv) | value_inv ⋅ (1 - value ⋅ value_inv) |
+    /// +----+-------+-----------+-----------------------+---------------------------------+-------------------------------------+
+    /// | V  | 0     | 0         | 1                     | 0
+    /// | 0                                   | |    | 0     | x         | 1
+    /// | 0                               | x                                   |
+    /// |    | x     | 0         | 1                     | x
+    /// | 0                                   | | V  | x     | 1/x       | 0
+    /// | 0                               | 0                                   |
+    /// |    | x     | y         | 1 - xy                | x(1 - xy)
+    /// | y(1 - xy)                           |
+    /// +----+-------+-----------+-----------------------+---------------------------------+-------------------------------------+
```

Comment:
@CPerezz the formatting seems broken here

Reply:
Ohhh damn!!! 
Let me make a quick fix PR